python/src/aiconfig/default_parsers/openai.py

```diff
@@ -74,40 +49,66 @@ def serialize(
         Returns:
             str: Serialized representation of the prompt and inference settings.
         """
-        # Serialize Prompt input
-        prompt_input = (
-            data["prompt"] if isinstance(data["prompt"], str) else {"data": data["prompt"]}
-        )
+        serialized_prompts = []
 
-        # Cosntruct model settings, does not include prompt
-        prompt_model_metadata = {key: value for key, value in data.items() if key != "prompt"}
+        # Combine conversation data with any extra keyword args
+        conversation_data = {**data} | kwargs
 
-        # check if config already has model settings in global metadata
-        model_name = prompt_model_metadata.get("model", self.id())
-        global_model_metadata = ai_config.metadata.models.get(model_name, {})
+        if not "messages" in conversation_data:
+            raise ValueError("Data must have `messages` array to match openai api spec")
 
-        if global_model_metadata:
-            # Check if the model settings from the input data are the same as the global model settings
-            # Compute the difference between the global model settings and the model settings from the input data
-            # If there is a difference, then we need to add the different model settings as overrides on the prompt's metadata
-            override_keys = set(prompt_model_metadata.keys()) - set(global_model_metadata.keys())
+        # Extract the global settings for the model
+        global_model_settings = ai_config.get_global_settings(self.id())
 
-            override_settings = {key: prompt_model_metadata[key] for key in override_keys}
+        if global_model_settings:
+            # Identify the settings that differ from global settings
+            override_settings = {
+                key: copy.deepcopy(conversation_data[key])
+                for key in conversation_data
+                if key not in global_model_settings
+            }
```

Comment:
This isn't comparing the values of the keys, so if the global settings contain `temperature` but set to a 0.9, while input completion params contained `temperature` but set to 1, this will not consider that to be an override even though it should be.

Reply:
Also check everywhere else in the code where setting overrides are being calculated, I am guessing the same problem exists elsewhere too.